src/components/Map/Map.tsx

```diff
@@ -10,14 +12,57 @@ interface IProps{
 }
 
 const Map: React.FC<IProps> = ( {countries} ) => {
+  const [selectedCountry, setSelectedCountry]= useState<any>('Select A Country')
+  
+  const [displayCountry, setDisplayCountry] = useState<boolean>(false)
+  const countryNames = countries.map(country => { 
+    return <option key={uuidv4()}>{country.name}</option>
+  })
+
+   // ********* Function to get chosen country for QuizPage component ********//
+   const getCurrentCountry = () => {
+   const country = countries.find(currCountry => currCountry.name.includes(selectedCountry))
+   return country;
+  }
+   // ******* Button click function  ********//
+   const handleSubmit = (e:any) => {
+   e.preventDefault()
+    if(getCurrentCountry()) {
```

Comment:
Is getCurrentCountry being invoked within the if statement? Would this work outside of the if statement?

Reply:
I see now that this is accounting for error handling if something goes wrong with the API!